src/frontend/context/DokumentutsendingContext.ts

```diff
@@ -183,6 +193,17 @@ export const [DokumentutsendingProvider, useDokumentutsending] = createUseContex
             nullstillSkjemaUtenomÅrsak();
         }, [årsak.verdi, bruker.status]);
 
+        useEffect(() => {
+            if (!toggles[ToggleNavn.verifiserDokdistKanal]) {
+                return;
+            }
+            if (bruker.status === RessursStatus.SUKSESS) {
+                hentDistribusjonskanal(bruker.data.personIdent);
+            } else {
+                nullstillDistribusjonskanal();
+            }
+        }, [bruker.status]);
```

Comment:
Siden vi uansett henter dette når bruker endrer seg, kunne vi ha lagt det på IPersonInfo(RestPersonInfo i backend)?

Reply:
Kan det bli behov for denne informasjonen fra andre steder i koden også, tror du?

Reply:
Vi må uansett ha personnummer, som vi ikke får før vi henter personen. Slik jeg tolker koden henter vi alltid distribusjonskanalen når vi henter brukeren. Det gjør at vi må ha to kall når vi kunne klart oss met ett. I tillegg gjør det frontentkoden mer kompleks. Nå har vi en useEffect som lytter på fagsak og som oppdaterer bruker når den endrer seg og så useEffecten på linje 205 oppdatere distrubisjonskanalen når brukeren oppdaterer seg. Det at vi kobler sammen useffecter på den måten gjør at det blir mer uoversiktlig å skjønne når ting oppdaterer seg og hvordan ting henger sammen

Reply:
Det virker som om useEffecten kun kjøres idet man navigerer til dokumentutsendingssiden, da. Men det kan sikkert skrives litt mer oversiktlig uansett, f.eks ved å gjøre hentingen inni en uavhengig useEffect i DokumentutsendingSkjema istedenfor

Reply:
Bedre nå, synes du? Henter den nå kun idet DokumentutsendingSkjema lastes inn..